core/src/main/java/org/elasticsearch/rest/action/fieldstats/RestFieldStatsAction.java

```diff
@@ -51,12 +52,16 @@ public RestFieldStatsAction(Settings settings, RestController controller, Client
     }
 
     @Override
-    public void handleRequest(final RestRequest request, final RestChannel channel, final Client client) {
+    public void handleRequest(final RestRequest request, final RestChannel channel, final Client client) throws Exception {
         final FieldStatsRequest fieldStatsRequest = new FieldStatsRequest();
         fieldStatsRequest.indices(Strings.splitStringByCommaToArray(request.param("index")));
         fieldStatsRequest.indicesOptions(IndicesOptions.fromRequest(request, fieldStatsRequest.indicesOptions()));
-        fieldStatsRequest.fields(Strings.splitStringByCommaToArray(request.param("fields")));
         fieldStatsRequest.level(request.param("level", FieldStatsRequest.DEFAULT_LEVEL));
+        if (RestActions.hasBodyContent(request)) {
+            fieldStatsRequest.source(RestActions.getRestContent(request));
+        } else {
+            fieldStatsRequest.setFields(Strings.splitStringByCommaToArray(request.param("fields")));
+        }
```

Comment:
What should we do if there are both a body and a `fields` parameter? Send an error maybe?